tests/platform_helper/providers/test_terraform_manifest.py

```diff
@@ -0,0 +1,170 @@
+
+    mock_echo_fn.assert_called_once_with("File created")
+
+    assert base_path == str(Path(".").absolute())
+    assert file_path == "terraform/codebase-pipelines/main.tf.json"
+    assert overwrite
+
+    json_content = json.loads(contents)
+
+    exp_version = version("dbt-platform-helper")
+    # Initial Comments
+    assert "WARNING: This is an autogenerated file, not for manual editing." in json_content["//"]
+    assert f"Generated by platform-helper {exp_version} / 2025-01-16 13:00:00" in json_content["//"]
+
+    local = json_content["locals"]
+    assert local["platform_config"] == '${yamldecode(file("../../platform-config.yml"))}'
+    assert local["application"] == '${local.platform_config["application"]}'
+    assert local["all_codebases"] == '${local.platform_config["codebase_pipelines"]}'
+    assert local["environments"] == '${local.platform_config["environments"]}'
+
+    aws_provider = json_content["provider"]["aws"]
+    assert aws_provider["region"] == "eu-west-2"
+    assert aws_provider["profile"] == "non-prod-acc"
+    assert aws_provider["alias"] == "non-prod-acc"
+    assert aws_provider["shared_credentials_files"] == ["~/.aws/config"]
+
+    terraform = json_content["terraform"]
+    assert terraform["required_version"] == SUPPORTED_TERRAFORM_VERSION
+
+    s3_backend = terraform["backend"]["s3"]
+    assert s3_backend["bucket"] == "terraform-platform-state-non-prod-acc"
+    assert s3_backend["key"] == "tfstate/application/my-app-codebase-pipelines.tfstate"
+    assert s3_backend["region"] == "eu-west-2"
+    assert s3_backend["encrypt"] is True
+    assert s3_backend["kms_key_id"] == "alias/terraform-platform-state-s3-key-non-prod-acc"
+    assert s3_backend["dynamodb_table"] == "terraform-platform-lockdb-non-prod-acc"
+
+    aws_req_provider = terraform["required_providers"]["aws"]
+    assert aws_req_provider["source"] == "hashicorp/aws"
+    assert aws_req_provider["version"] == SUPPORTED_AWS_PROVIDER_VERSION
+
+    module = json_content["module"]["codebase-pipelines"]
+    assert (
+        module["source"]
+        == f"git::https://github.com/uktrade/terraform-platform-modules.git//codebase-pipelines?depth=1&ref=7"
+    )
+    assert module["for_each"] == "${local.all_codebases}"
+    assert module["application"] == "${local.application}"
+    assert module["codebase"] == "${each.key}"
+    assert module["repository"] == "${each.value.repository}"
+    assert (
+        module["additional_ecr_repository"]
+        == '${lookup(each.value, "additional_ecr_repository", null)}'
+    )
+    assert module["pipelines"] == "${each.value.pipelines}"
+    assert module["services"] == "${each.value.services}"
+    assert module["requires_image_build"] == '${lookup(each.value, "requires_image_build", true)}'
+    assert (
+        module["slack_channel"]
+        == '${lookup(each.value, "slack_channel", "/codebuild/slack_channel_id")}'
+    )
+    assert module["env_config"] == "${local.environments}"
```

Comment:
Love the many, specific asserts. Will make it very easy to see where things go wrong, if they do.

Reply:
Much better than the templated manifests where we do a shotgun text search on some things.